specs/SharedBuffer.md

```diff
@@ -0,0 +1,333 @@
+  HRESULT Close();
+}
+
+interface ICoreWebView2_14 : IUnknown {
+  /// Share a shared buffer object with script of the main frame in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object as the result of parsing `additionalDataAsJson` as JSON string.
+  ///           This property will be `undefined` if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If a string is provided as `additionalDataAsJson` but it is not a valid JSON string,
+  /// the API will fail with `E_INVALIDARG`.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access
+  /// violation in WebView renderer process and crash the renderer process.
+  /// If the shared buffer is already closed, the API will fail with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  /// 
+  /// The script code should call `chrome.webview.releaseBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access to the shared buffer any more.
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
+  /// If a buffer is shared to a untrusted site, possible sensitive information could be leaked.
+  /// If a buffer is shared as modifiable by the script and the script modifies it in an unexpected way,
+  /// it could result in corrupted data that might even crash the application. 
+  HRESULT PostSharedBufferToScript(
+    [in] ICoreWebView2SharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+interface ICoreWebView2Frame4 : IUnknown {
+  /// Share a shared buffer object with script of the iframe in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object as the result of parsing `additionalDataAsJson` as JSON string.
+  ///           This property will be `undefined` if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If a string is provided as `additionalDataAsJson` but it is not a valid JSON string,
+  /// the API will fail with `E_INVALIDARG`.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access
+  /// violation in WebView renderer process and crash the renderer process.
```

Comment:
You are absolutely right. It is very desirable to throw a JS exception instead of crashing the renderer process.
However, there is currently no read-only ArrayBuffer concept/implementation in JS. So, we have to accept this crashing behavior.
We have a task to look into whether we could support readonly ArrayBuffer in the future. If that is achieved, we will switch the implementation and update the documentation. It is not an easy task. It has been requested in the community once a while for years, and still not implemented by anyone.

Reply:
No change here. We'll continue working with Edge folks to see if we can do better for this in the future.